torchgeo/datasets/bigearthnet.py

```diff
@@ -572,3 +582,444 @@ def plot(
+
+        tensor = torch.from_numpy(map)
+        # remap to ordinal values
+        for corine, ordinal in self.clc_codes.items():
+            tensor[tensor == corine] = ordinal
+        return tensor.long()
+
+    def _load_target(self, index: int) -> Tensor:
+        """Load the target mask for a single image.
+
+        Args:
+            index: index to return
+
+        Returns:
+            the target label
+        """
+        label_names = self.metadata_df.iloc[index]['labels']
+
+        indices = [self.class2idx[label_names] for label_names in label_names]
+
+        image_target = torch.zeros(self.num_classes, dtype=torch.long)
+        image_target[indices] = 1
+        return image_target
+
+    def _verify(self) -> None:
+        """Verify the integrity of the dataset."""
+        exists = []
+        for key, metadata in self.metadata_locs.items():
+            if key != 'metadata':
+                exists.append(
+                    os.path.exists(os.path.join(self.root, metadata['directory']))
+                )
+            else:
+                exists.append(
+                    os.path.exists(os.path.join(self.root, metadata['filename']))
+                )
+
+        if all(exists):
+            return
+
+        # check if compressed files already exist
+        exists = []
+        for key, metadata in self.metadata_locs.items():
+            if key == 'metadata':
+                if os.path.exists(os.path.join(self.root, metadata['filename'])):
+                    exists.append(True)
+                else:
+                    exists.append(False)
+            else:
+                for fname in metadata['files']:
+                    fpath = os.path.join(self.root, fname)
+                    if os.path.exists(fpath):
+                        exists.append(True)
+                    else:
+                        exists.append(False)
+
+        if all(exists):
+            return
+
+        if not self.download:
+            raise DatasetNotFoundError(self)
+
+        self._download()
+        self._extract()
+
+    def _download(self) -> None:
+        """Download the required tarball parts using the URL template and md5 sums."""
+        for key, meta in self.metadata_locs.items():
+            if key == 'metadata':
+                continue
+            for fname, md5 in meta['files'].items():
+                target_path = os.path.join(self.root, fname)
+                if not os.path.exists(target_path):
+                    download_url(self.url.format(fname), target_path, md5)
+
+    def _extract(self) -> None:
+        """Extract the tarball parts.
+
+        For each modality (s1, s2, maps), its parts are concatenated together and then extracted.
+        """
+        chunk_size = 2**15  # same as used in torchvision and ssl4eo
+        for key, meta in self.metadata_locs.items():
+            if key == 'metadata':
+                continue
+            assert not os.path.isdir(os.listdir(self.root)[2])
```

Comment:
@adamjstewart any idea what could cause this error? Basically, when you check whether the tar ball file in the root is a directory it says no, but in the next line, when you do `os.path.join()` with the root, it all of the sudden is a directory... I don't see what I messed up.

Reply:
Fixed in latest commit.